java-generator/core/src/main/java/io/fabric8/java/generator/CRGeneratorRunner.java

```diff
@@ -45,48 +47,47 @@ public void run(File source, File basePath) {
     }
 
     public List<WritableCRCompilationUnit> generate(
-            CustomResourceDefinition crd, Optional<String> basePackageName) {
+            CustomResourceDefinition crd, String basePackageName) {
         CustomResourceDefinitionSpec crSpec = crd.getSpec();
         String crName = crSpec.getNames().getKind();
         String group = crSpec.getGroup();
 
-        List<WritableCRCompilationUnit> writableCUs =
-                new ArrayList<WritableCRCompilationUnit>(crSpec.getVersions().size());
+        List<WritableCRCompilationUnit> writableCUs = new ArrayList<>(crSpec.getVersions().size());
         for (CustomResourceDefinitionVersion crdv : crSpec.getVersions()) {
             CompilationUnit cu = new CompilationUnit();
 
             String version = crdv.getName();
 
-            String pkg = basePackageName.map((p) -> p + "." + version).orElse(version);
+            String pkg = null;
```

Comment:
You could have kept the `Optional` version by doing `Optional.ofNullable(basePackageName)...`

Reply:
You asked for a simplification and I simplified it 😛 , no real need to use an `Optional` here, as you mentioned the `packageName` should be always present.

Reply:
doesn't matter, fixed